core-ui/src/apollo.js

```
@@ -117,26 +109,41 @@ export function createKymaApolloClient() {
   });
 }
 
+export const ApolloClientProvider = ({ children, createClient, provider }) => {
+  const context = useMicrofrontendContext();
+  const { fromConfig } = useConfig();
+  const [client, setClient] = useState(null);
+
+  useEffect(() => {
+    if (!context.idToken) {
+      return;
+    }
+    const client = createClient(fromConfig, context.idToken);
+    setClient(client);
+    return () => {
+      try {
+        client && client.stop();
+      } finally {
```

I don't get why there's the `try` block here. In case `client.stop()` throws an error there will be an uncaught exception anyway.

---

No, in this case, an exception will be swallowed. During first render client is invalid and it will always throw an exception on second render unless it is wrapped in `try`. If you have a better idea of how to handle it I'm open.

---

Actually, you're right. Any exception will be uncaught. It must have gotten fixed because of something else. I'll check it out again.